patches/tModLoader/Terraria/ModLoader/UI/ModBrowser/UIModBrowser.cs

```diff
@@ -501,6 +504,64 @@ internal static async Task<bool> DownloadMods(IEnumerable<ModDownloadItem> mods,
 		}
 	}
 
+	internal static void DownloadModsNoUI(IEnumerable<ModDownloadItem> mods)
+	{
+		var set = mods.ToHashSet();
+		Interface.modBrowser.SocialBackend.GetDependenciesRecursive(set); // Causes a bit of lag
+
+		var fullList = ModDownloadItem.NeedsInstallOrUpdate(set).ToList();
+		if (!fullList.Any())
+			return;
+
+		lock (DownloadQueue) {
+			foreach (var item in fullList) {
+				DownloadQueue.Enqueue(item);
+
+				var notification = new ModDownloadNotification(item);
+				InGameNotificationsTracker.AddMenuNotification(notification);
+				item.notification = notification;
+			}
+		}
+
+		lock (Interface.modBrowser.modSlugsToUpdateInstallInfo) {
+			Interface.modBrowser.modSlugsToUpdateInstallInfo.UnionWith(fullList.Select(x=>x.ModName));
+		}
+
+		if (DownloadTask == null || DownloadTask.IsCompleted) {
+			DownloadTask = Task.Run(DownloadTaskTask);
+		}
+	}
+
+	internal static void DownloadTaskTask()
+	{
+		try {
+			while (DownloadQueue.Count > 0) {
+				var mod = DownloadQueue.Dequeue();
+
+				bool wasInstalled = mod.IsInstalled;
+
+				if (ModLoader.TryGetMod(mod.ModName, out var loadedMod)) {
+					loadedMod.Close();
+
+					// We must clear the Installed reference in ModDownloadItem to facilitate downloading, in addition to disabling - Solxan
+					mod.Installed = null;
+					//setReloadRequred?.Invoke();
+				}
+				 
+				Interface.modBrowser.SocialBackend.DownloadItem(mod, mod.notification);
+				//if (!wasInstalled)
+				//	onNewModInstalled?.Invoke(mod);
+
+				ModOrganizer.LocalModsChanged([mod.ModName], isDeletion: false);
+			}
+		}
+		catch (Exception ex) {
+			//LogModBrowserException(ex, previousMenuId);
```

Comment:
So... this is the part to think about. What happens to the rest of the `DownloadQueue`? Do we report the error to the user?

Reply:
Could provide a red/error notification?